src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -469,8 +469,33 @@ public bool TryGetDateTime(out DateTime value)
                 throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
             }
 
-            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
+            ReadOnlySpan<byte> span = ValueSpan;
+
+            if (HasValueSequence)
+            {
+                JsonReaderHelper.GetValueSequenceAsSpan(ValueSequence, out span);
+            }
+
+            int bytesConsumed;
+
+            if (_stringHasEscaping)
+            {
+                int idx = span.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(idx != -1);
+
+                Span<byte> utf8Unescaped = stackalloc byte[span.Length];
```

Comment:
This is an unconstrained stackalloc. It should be constrained, just like the `TryGetGuid`.